packages/core/src/Parser.ts

```diff
@@ -222,61 +220,72 @@ export class Parser {
-            nextContext.currentNode = node.childNodes[0];
+            context.currentNode = node.childNodes[0];
             // Text node cannot have children, therefore `node` is an Element, not
             // a text node. Only text nodes can be represented by multiple VNodes,
             // so the first matching VNode can safely be selected from the map.
             // If `node` as no VNode representation  in the map (e.g. format nodes),
             // its children are added into the current `parentVNode`.
             const parsedParent = VDocumentMap.fromDom(node);
             if (parsedParent) {
-                nextContext.parentVNode = parsedParent[0];
+                context.parentVNode = parsedParent[0];
             }
-            this._contextStack.push(nextContext);
+            this._contextStack.push(context);
         } else if (node.nextSibling) {
             // Parse the siblings of the current node with the same parent, if any.
-            nextContext.currentNode = node.nextSibling;
+            this.currentContext.currentNode = node.nextSibling;
         } else {
             // Parse the next ancestor sibling in the ancestor tree, if any.
             let ancestor = node;
             // Climb back the ancestor tree to the first parent having a sibling.
-            const rootNode = this.currentContext.rootNode;
-            do {
+            const rootNode = context.rootNode;
+            while (ancestor && !ancestor.nextSibling && ancestor !== rootNode) {
                 ancestor = ancestor.parentNode;
-                nextContext = this._contextStack.pop();
                 if (ancestor && Format.tags.includes(ancestor.nodeName)) {
                     // Pop last formatting context from the stack
-                    nextContext.format.shift();
+                    this.currentContext.format.shift();
                 }
-            } while (ancestor && !ancestor.nextSibling && ancestor !== rootNode);
+                this._contextStack.pop();
+            }
             // At this point, the found ancestor has a sibling.
             if (ancestor && ancestor !== rootNode) {
-                // Text node cannot have children, therefore parent is an Element,
-                // not a text node. Only text nodes can be represented by multiple
-                // VNodes so the first VNode can safely be selected from the map.
-                nextContext.currentNode = ancestor.nextSibling;
+                if (!this.currentContext) {
+                    // If we went back up the whole stack, we start with a new
+                    // context.
+                    this._contextStack.push({
+                        rootNode: context.rootNode,
+                        currentNode: ancestor.nextSibling,
+                        parentVNode: context.parentVNode,
+                        format: [],
+                        vDocument: context.vDocument,
+                    });
```

Comment:
What ? There was an initial context at the beginning when the parsing started. There is no reason this one could not be retrieved here.